src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -680,8 +694,9 @@ private static bool HasApplicableConditionalMethod(OverloadResolutionResult<Meth
 
             var validResult = resolution.OverloadResolutionResult.ValidResult;
             var args = resolution.AnalyzedArguments.Arguments.ToImmutable();
+            var refKindsArray = resolution.AnalyzedArguments.RefKinds.ToImmutableOrNull();
 
-            ReportBadDynamicArguments(syntax, args, diagnostics, queryClause);
+            ReportBadDynamicArguments(syntax, args, refKindsArray, diagnostics, queryClause);
```

Comment:
>refKindsArray [](start = 52, length = 13)

Not sure if this is the right thing to do. It looks like we are not doing dynamic dispatch for local functions. Please add a test.

Reply:
We perform other checks. I guess - at least for consistency with regular functions.

Will add a test.

---
In reply to: [161102102](https://github.com/dotnet/roslyn/pull/24164#discussion_r161102102) [](ancestors = 161102102)